lib/mergeArrayByName.js

```diff
@@ -0,0 +1,34 @@
+// https://github.com/KyleAMathews/deepmerge#arraymerge
+
+const merge = require('deepmerge')
+
+function findMatchingIndex (sourceItem, target) {
+  var matchingIndex
+  if (Object.keys(sourceItem).includes('name')) {
```

Comment:
Could use `Object.hasOwnProperty` to be just a tiny bit cleaner:

```suggestion
  if (sourceItem.hasOwnProperty('name')) {
```